Java3practise4/J2_Stream201126/chat-client/src/ru/geekbrains/java_two/chat/client/ChangeLoginFrame.java

```diff
@@ -0,0 +1,140 @@
+
+
+    private final JButton btnDisconnect = new JButton("<html><b>Disconnect</b></html>");
+    private final JTextField tfMessage = new JTextField();
+    private final JButton btnSend = new JButton("Send");
+
+    private final JList<String> userList = new JList<>();
+    private boolean shownIoErrors = false;
+    private SocketThread socketThread;
+
+    private final JMenuBar menuBar = new JMenuBar();
+    private final JMenu fileMenu = new JMenu("Menu");
+    private final JMenuItem changeLogin = new JMenuItem("Change login");
+
+
+    public ChangeLoginFrame(ClientGUI clientGUI) {
+
+        Thread.setDefaultUncaughtExceptionHandler(this);
+        setDefaultCloseOperation(WindowConstants.HIDE_ON_CLOSE);
+        setLocationRelativeTo(null);
+        setSize(WIDTH, HEIGHT);
+        setTitle(WINDOW_TITLE);
+
+        btnCancel.addActionListener(this);
+        btnChange.addActionListener(this);
+
+
+        Font font = new Font("Verdana", Font.PLAIN, 11);
+        fileMenu.setFont(font);
+        changeLogin.setFont(font);
+        fileMenu.add(changeLogin);
+        menuBar.add(fileMenu);
+
+        setJMenuBar(menuBar);
+
+        changeLogin.addActionListener(this);
+
+        setPreferredSize(new Dimension(270, 225));
+        pack();
+        setLocationRelativeTo(null);
+        setVisible(true);
+
+//        panelTop.add(tfIPAddress);
+//        panelTop.add(tfPort);
+//        panelTop.add(cbAlwaysOnTop);
+//        panelTop.add(tfLogin);
+//        panelTop.add(tfPassword);
+//        panelTop.add(btnLogin);
+//
+//
+//        panelBottom.add(btnDisconnect, BorderLayout.WEST);
+//        panelBottom.add(tfMessage, BorderLayout.CENTER);
+//        panelBottom.add(btnSend, BorderLayout.EAST);
+//
+//        panelBottom.setVisible(false);
+
+        panelTextFields.add(tfLogin);
+        panelTextFields.add(tfLoginNew);
+        panelTextFields.add(tfPassword);
+
+        panelButtons.add(btnChange);
+        panelButtons.add(btnCancel);
+
+
+
+        add(panelTextFields, BorderLayout.NORTH);
+        add(panelButtons, BorderLayout.SOUTH);
+
+
+
+        setVisible(false);
+
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        Object src = e.getSource();
+
+        if  (src == btnSend || src == tfMessage) {
+          //  sendMessage();
+        } else if (src == btnChange) {
+          //  connect();
+        } else if (src == btnCancel) {
+          //  socketThread.close();
+        } else if (src == changeLogin) {
+         //   changeLogin();
+        } else {
+            throw new RuntimeException("Undefined source: " + src);
+        }
+    }
+
+    private void showException(Thread t, Throwable e) {
+        String msg;
+        StackTraceElement[] ste = e.getStackTrace();
+        if (ste.length == 0)
+            msg = "Empty Stacktrace";
+        else {
+            msg = String.format("Exception in thread \"%s\" %s: %s\n\tat %s",
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.